File: src/TypeScript/src/ghom/objects/OrganizationInvitation.ts
import { ID } from "../scalars/Id";
import { OrganizationInvitationType } from "../enums/OrganizationInvitationType";
import { User } from "./User";
import { Organization } from "./Organization";
import { OrganizationInvitationRole } from "../enums/OrganizationInvitationRole";
import { Node } from "../interfaces/Node";

// OrganizationInvitation
// An Invitation for a user to an organization.

// Implements
// Node
// Fields
// createdAt (DateTime!)
// Identifies the date and time when the object was created.

// email (String)
// The email address of the user invited to the organization.

// id (ID!)
// invitationType (OrganizationInvitationType!)
// The type of invitation that was sent (e.g. email, user).

// invitee (User)
// The user who was invited to the organization.

// inviter (User!)
// The user who created the invitation.

// organization (Organization!)
// The organization the invite is for

// role (OrganizationInvitationRole!)
// The user's pending role in the organization (e.g. member, owner).





// An Invitation for a user to an organization.
export class OrganizationInvitation implements Node {
    
    // Fields

    // Identifies the date and time when the object was created.
    createdAt: Date

    // The email address of the user invited to the organization.
    email: String

    id: ID

    // The type of invitation that was sent (e.g. email, user).
    invitationType: OrganizationInvitationType

    // The user who was invited to the organization.
    invitee: User

    // The user who created the invitation.
    inviter: User

    // The organization the invite is for
    organization: Organization

    // The user's pending role in the organization (e.g. member, owner).
    role: OrganizationInvitationRole
}

File: TypeScript/resources/code/GitHub/oktokit/octokit.graphql.net/Octokit.GraphQL/Model/OrganizationInvitation.cs
namespace Octokit.GraphQL.Model
{
    using System;
    using System.Collections.Generic;
    using System.Linq.Expressions;
    using Octokit.GraphQL.Core;
    using Octokit.GraphQL.Core.Builders;

    /// <summary>
    /// An Invitation for a user to an organization.
    /// </summary>
    public class OrganizationInvitation : QueryableValue<OrganizationInvitation>
    {
        internal OrganizationInvitation(Expression expression) : base(expression)
        {
        }

        /// <summary>
        /// Identifies the date and time when the object was created.
        /// </summary>
        public DateTimeOffset CreatedAt { get; }

        /// <summary>
        /// The email address of the user invited to the organization.
        /// </summary>
        public string Email { get; }

        public ID Id { get; }

        /// <summary>
        /// The type of invitation that was sent (e.g. email, user).
        /// </summary>
        public OrganizationInvitationType InvitationType { get; }

        /// <summary>
        /// The user who was invited to the organization.
        /// </summary>
        public User Invitee => this.CreateProperty(x => x.Invitee, Octokit.GraphQL.Model.User.Create);

        /// <summary>
        /// The user who created the invitation.
        /// </summary>
        public User Inviter => this.CreateProperty(x => x.Inviter, Octokit.GraphQL.Model.User.Create);

        /// <summary>
        /// The organization the invite is for
        /// </summary>
        public Organization Organization => this.CreateProperty(x => x.Organization, Octokit.GraphQL.Model.Organization.Create);

        /// <summary>
        /// The user's pending role in the organization (e.g. member, owner).
        /// </summary>
        public OrganizationInvitationRole Role { get; }

        internal static OrganizationInvitation Create(Expression expression)
        {
            return new OrganizationInvitation(expression);
        }
    }
}